lib/internal/modules/esm/module_job.js

```diff
@@ -72,9 +72,16 @@ const isCommonJSGlobalLikeNotDefinedError = (errorMessage) =>
  * @param {string} url
  * @returns {void}
  */
-const explainCommonJSGlobalLikeNotDefinedError = (e, url) => {
+const explainCommonJSGlobalLikeNotDefinedError = (e, url, hasTopLevelAwait) => {
   if (e?.name === 'ReferenceError' &&
       isCommonJSGlobalLikeNotDefinedError(e.message)) {
+
+    if (hasTopLevelAwait) {
+      e.message = `ERR_AMBIGUOUS_MODULE_SYNTAX: This file cannot be parsed as either CommonJS or ES Module. CommonJS error: await is only valid in async functions. ES Module error: require is not defined in ES module scope. If you meant to use CommonJS, wrap top-level await in async function. If you meant to use ESM, do not use require().`;
```

Comment:
Okay, I'm a little confused here -- the code mostly looks good, but maybe we can spell out which condition exactly is supposed to trigger this error, and be specific about the error message? Like, it's odd to describe multiple different errors here, although I kind of see where you're coming from

Reply:
Hi @addaleax, 

You're absolutely right - the message is trying to do too much. 

The specific condition is: file has both `require()` and top-level `await`, creating an ambiguous parsing situation.

How about we simplify the error message to focus on the immediate issue:

```bash
ERR_AMBIGUOUS_MODULE_SYNTAX: Cannot use 'require()' and top-level 'await' in the same file.
Choose either CommonJS (remove await, use async functions) or ES Module (remove require, use import).
```

Reply:
Technically it can be _parsed_ as ESM, just not successfully _evaluated_ as ESM because `require` is undefined. It can't be successfully parsed as CommonJS because of the top-level await. So I'm not sure if "ambiguous" is the right word; it's more that it's so unclear what the user intended that we can't reasonably make a guess.

Reply:
> Technically it can be _parsed_ as ESM, just not successfully _evaluated_ as ESM because `require` is undefined. It can't be successfully parsed as CommonJS because of the top-level await. So I'm not sure if "ambiguous" is the right word; it's more that it's so unclear what the user intended that we can't reasonably make a guess.

Yes, I agree with what you said, I think a message like this would be more appropriate.

```js
Cannot determine intended module format. File contains both 'require()' and top-level 'await'. For CommonJS: wrap await in async function. For ES Module: replace require() with import.
```

Reply:
Thanks I updated